### PR TITLE
make URL field optional for themes

### LINKED ISSUE #4353: Avoid URL field for themes
The URL field should be hidden (and not required) for themes.

grafik

cc @gfr

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | fixes #4353 
| Refs tickets      | -
| License           | MIT
| Changelog updated | yes

## Description

This PR adds a form option for extension editing to make the URL field optional for themes.

### CODE REVIEW

File: src/system/ExtensionsModule/Form/Type/ExtensionModifyType.php [hunk 1]
```diff
@@ -73,6 +74,10 @@ public function getBlockPrefix()
 
     public function configureOptions(OptionsResolver $resolver)
     {
+        $resolver->setRequired('extensionType');
+        $resolver->setAllowedTypes('extensionType', 'string');
+        $resolver->setAllowedValues('extensionType', ['module', 'theme']);
+
         $resolver->setDefaults([
             'data_class' => ExtensionEntity::class
```
Comment (craigh) [inline src/system/ExtensionsModule/Form/Type/ExtensionModifyType.php:82]: what about adding default = module?

Comment (Guite) [reply src/system/ExtensionsModule/Form/Type/ExtensionModifyType.php:83]: changed in 8497b909e99126073611575b84a3147c0ac78551

Comment (craigh) [reply src/system/ExtensionsModule/Form/Type/ExtensionModifyType.php:83]: interesting. I thought it belonged in the `setDefaults` method. looks fine.

Comment (Guite) [reply src/system/ExtensionsModule/Form/Type/ExtensionModifyType.php:83]: oh yes, this is even better; will move it

Comment (Guite) [reply src/system/ExtensionsModule/Form/Type/ExtensionModifyType.php]: changed in 9f0f0de281d68b18dc178c76f751559f77678e45

**Pull request merged**